# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/nsCocoaWindow.mm
# Commit: 5791073c28b7
# Full Hash: 5791073c28b727964776360a5b4dfd5c004ce152
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2021-04-13 03:56:47
# Description:
#   Bug 1703762 - Delegate StartAsyncAutoscroll and StopAsyncAutoscroll calls to the proper widget mPopupContentView. r=botond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D111575
# ==============================================================================

diff -r 4a25e6b0bf09 -r 5791073c28b7 widget/cocoa/nsCocoaWindow.mm
--- a/widget/cocoa/nsCocoaWindow.mm	Mon Apr 12 22:50:22 2021 +0000
+++ b/widget/cocoa/nsCocoaWindow.mm	Mon Apr 12 23:05:40 2021 +0000
@@ -2572,6 +2572,22 @@
   return nsBaseWidget::AsyncPanZoomEnabled();
 }
 
+bool nsCocoaWindow::StartAsyncAutoscroll(const ScreenPoint& aAnchorLocation,
+                                         const ScrollableLayerGuid& aGuid) {
+  if (mPopupContentView) {
+    return mPopupContentView->StartAsyncAutoscroll(aAnchorLocation, aGuid);
+  }
+  return nsBaseWidget::StartAsyncAutoscroll(aAnchorLocation, aGuid);
+}
+
+void nsCocoaWindow::StopAsyncAutoscroll(const ScrollableLayerGuid& aGuid) {
+  if (mPopupContentView) {
+    mPopupContentView->StopAsyncAutoscroll(aGuid);
+    return;
+  }
+  nsBaseWidget::StopAsyncAutoscroll(aGuid);
+}
+
 already_AddRefed<nsIWidget> nsIWidget::CreateTopLevelWindow() {
   nsCOMPtr<nsIWidget> window = new nsCocoaWindow();
   return window.forget();
